src/Web/Web.csproj

```diff
@@ -38,7 +35,12 @@
 		<Exec WorkingDirectory="$(SpaRoot)\" Command="npm install" />
 	</Target>
 
-	<Target Name="PublishRunWebpack" BeforeTargets="Build" DependsOnTargets="NpmInstall">
+	<Target Name="NpmRunBuild" AfterTargets="NpmInstall">
+		<Exec WorkingDirectory="$(SpaRoot)\" Command="npm run build"/>
+	</Target>
+
+	<Target Name="PublishRunWebpack" AfterTargets="ComputeFilesToPublish">
+		<Exec WorkingDirectory="$(SpaRoot)\" Command="npm install" />
```

Comment:
Don't we want to have this depend on the `NpmInstall` target? That way we aren't duplicating logic. For example this target does not check for the presence of `node` on the system whereas the `NpmInstall` target does.

Reply:
The problem with depending on the `NpmInstall` target is that it only triggers when there is no node_modules, which is not enough. If we update / add a npm library it will not trigger and it will not install the incoming changes, resulting in an error. As an alternative, we can remove the condition and make this depend on npminstall.

Reply:
We could extend the condition to check if package.json has been changed. That would handle both cases:

1. We want to run `npm install` any time new packages are introduced
2. We do NOT want to run `npm install` if we already have everything installed.

```
<Target Name="NpmInstall" Inputs="$(SpaRoot)\package.json" Outputs="node_modules/.install-stamp">
    ...
    <!-- Write the stamp file, so incremental builds work -->
    <Touch Files="node_modules/.install-stamp" AlwaysCreate="true" />
</Target>
```

Reply:
Ok, will look into doing that